proposals/interface-types/working-notes/record.md

```diff
@@ -0,0 +1,73 @@
+
+
+### Records
+
+Records are collections of named and typed fields. Records can be defined in as
+Interface Types; for example this definition encodes a familiar notion of a
+credit card:
+
+```wat
+(@interface datatype @cc 
+  (record "cc"
+    (ccNo u64)
+    (name string)
+    (expires (record
+      (mon u8)
+      (year u16)
+      )
+    )
+    (ccv u16)
+  )
+)
+```
+
+Lifting and lowering a record involves lifting/lowering the individual fields of
+the record and then 'packing' or 'unpacking' the record as an ordered
+sub-sequence of elements on the stack.
+
+For example, the sequence:
+
+```wat
+...
+local.get $ccNo
+i64-to-u64
+local.get $name
+local.get $name.len
+memory-to-string
+i8.local.get $mon
+i8_to_u8
+i16.local.get $year
+i16_to_u16
+pack @(record (mon u8) (year u16))
```

Comment:
Could `pack` be written `pack $cc` (adopting the abovementioned suggestion to replace `@cc` with `$cc`)?

Reply:
Yes. In this case though, the record is a structural type; I guess that we could require structural types to be defined too ...
The replacement would either be:
pack (record (field "mon" u8) (field "year" u16))
or 
(@interface type $ccExpiry 
  (record (field "mon" u8) (field "year" u16)))
...
pack $ccExpiry
Personally am open to either style